zebrad/src/components/inbound.rs

```diff
@@ -73,61 +73,62 @@ impl Service<zn::Request> for Inbound {
 
     #[instrument(skip(self, cx))]
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        use oneshot::error::TryRecvError;
-        match self.network_setup.take() {
-            // If we're waiting for setup, check if we became ready
-            Some(mut rx) => match rx.try_recv() {
+        // Check whether the network setup is finished, but don't wait for it to
+        // become ready before reporting readiness. We expect to get it "soon",
+        // and reporting unreadiness might cause unwanted load-shedding, since
+        // the load-shed middleware is unable to distinguish being unready due
+        // to load from being unready while waiting on setup.
```

Comment:
Bummer

Reply:
On the other hand, doing this means that we can start responding to state queries immediately.